Realm/ObjectStore/object_store.cpp

```diff
@@ -487,6 +516,57 @@ void ObjectStore::delete_data_for_object(Group *group, StringData object_type) {
     }
 }
 
+void ObjectStore::rename_column(Group *group, Schema& passed_schema, StringData object_type, StringData old_name, StringData new_name) {
+    Schema schema = schema_from_group(group);
+    auto matching_schema = schema.find(object_type);
+    if (matching_schema == schema.end()) {
+        throw PropertyRenameMissingOldObjectTypeException(object_type);
+    }
+    Property *old_property = matching_schema->property_for_name(old_name);
+    if (old_property == nullptr) {
+        throw PropertyRenameMissingOldPropertyException(old_name, new_name);
+    }
+    TableRef table = table_for_object_type(group, object_type);
+    if (!table) {
+        throw PropertyRenameMissingNewObjectTypeException(object_type);
+    }
+    auto passed_object_schema = passed_schema.find(object_type);
+    if (passed_object_schema == passed_schema.end()) {
+        throw PropertyRenameMissingNewObjectTypeException(object_type);
+    }
+    Property *new_property = matching_schema->property_for_name(new_name);
+    if (new_property == nullptr) {
+        throw PropertyRenameMissingNewPropertyException(old_name, new_name);
+    }
+    if (old_property->type != new_property->type ||
+        old_property->object_type != new_property->object_type) {
+        throw PropertyRenameTypeMismatchException(*old_property, *new_property);
+    }
+    if (passed_object_schema->property_for_name(old_name) != nullptr) {
+        throw PropertyRenameOldStillExistsException(old_name, new_name);
+    }
+    size_t column_to_remove = new_property->table_column;
+    table->rename_column(old_property->table_column, new_name);
+    table->remove_column(column_to_remove);
```

Comment:
This changes the column numbers for any other newly-added columns after the one being removed, which doesn't appear to be handled.